linkerd/app/outbound/src/lib.rs

```diff
@@ -605,3 +598,7 @@ fn is_discovery_rejected(err: &Error) -> bool {
     tracing::debug!(rejected, %err);
     rejected
 }
+
+fn is_loop(err: &(dyn std::error::Error + 'static)) -> bool {
+    err.is::<prevent_loop::LoopPrevented>() || err.source().map(is_loop).unwrap_or(false)
+}
```

Comment:
nit/tioli: `is_discovery_rejected` has a `debug!` event in it, should this as well?